src/app/organization/offices/edit-office/edit-office.component.ts

```diff
@@ -0,0 +1,84 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { DatePipe } from '@angular/common';
+
+/** Custom Services */
+import { OrganizationService } from 'app/organization/organization.service';
+
+/**
+ * Edit Office component.
+ */
+@Component({
+  selector: 'mifosx-edit-office',
+  templateUrl: './edit-office.component.html',
+  styleUrls: ['./edit-office.component.scss']
+})
+export class EditOfficeComponent implements OnInit {
+
+  /** Selected Data. */
+  officeData: any;
+  /** Office form. */
+  officeForm: FormGroup;
+  /** Date. */
+  openedOn: any;
+
+    /**
+     * Retrieves the charge data from `resolve`.
+     * @param {ProductsService} organizationService Organization Service.
+     * @param {FormBuilder} formBuilder Form Builder.
+     * @param {ActivatedRoute} route Activated Route.
+     * @param {Router} router Router for navigation.
+     * @param {MatDialog} dialog Dialog reference.
+     * @param {DatePipe} datepipe Convert Date.
+     */
+    constructor(private organizationService: OrganizationService,
+                private formBuilder: FormBuilder,
+                private route: ActivatedRoute,
+                private router: Router,
+                private datepipe: DatePipe) {
+      this.route.data.subscribe((data: { officeTemplate: any }) => {
+        this.officeData = data.officeTemplate;
+        this.openedOn = `${this.officeData.openingDate[0]}-${this.officeData.openingDate[1]}-${this.officeData.openingDate[2]}`;
+      });
+    }
+
+  ngOnInit() {
+    this.createOfficeForm();
+  }
+
+  /**
+   * Create Edit Office Form.
+   */
+  createOfficeForm() {
+    this.officeForm = this.formBuilder.group({
+      'name': [this.officeData.name, Validators.required],
+      'parentId': [this.officeData.parentName, Validators.required],
+      'openingDate': [this.openedOn, Validators.required],
```

Comment:
Look at line 68 in #604 it has similar implementation to openedOn date you are trying to do.(ie: an array[2])
```suggestion
```